lib/ansible/modules/monitoring/uptimerobot.py

```diff
@@ -27,124 +20,168 @@
+    check_type: http
     apikey: 12345-1234512345
-    state: started
 '''
 
-import json
-
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.six.moves.urllib.parse import urlencode
-from ansible.module_utils.urls import fetch_url
 
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
 
-API_BASE = "https://api.uptimerobot.com/"
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import Request
 
-API_ACTIONS = dict(
-    status='getMonitors?',
-    editMonitor='editMonitor?'
-)
+API_BASE = "https://api.uptimerobot.com/v2/"
 
 API_FORMAT = 'json'
 API_NOJSONCALLBACK = 1
 CHANGED_STATE = False
 SUPPORTS_CHECK_MODE = False
 
 
-def checkID(module, params):
-
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['status'] + data
-    req, info = fetch_url(module, full_uri)
-    result = req.read()
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult
-
-
-def startMonitor(module, params):
-
-    params['monitorStatus'] = 1
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['editMonitor'] + data
-    req, info = fetch_url(module, full_uri)
-    result = req.read()
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
-
-
-def pauseMonitor(module, params):
-
-    params['monitorStatus'] = 0
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['editMonitor'] + data
-    req, info = fetch_url(module, full_uri)
-    result = req.read()
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
+class UptimeRobot:
+    def __init__(self, params, api_methods):
+        self.headers = {}
+        self.headers['Content-Type'] = "application/json"
+        self.uri = ''
+        self.api_methods = api_methods
+        self.params = params
+        self.params = self.check_dict(self.params)
+        for k, v in self.params.items():
+            if k == 'type':
+                if v == 'dns':
+                    self.params[k] = 2
+                else:
+                    self.params[k] = 1
+        self.body = json.dumps(self.params)
+        for state, method in self.api_methods.items():
+            if state == self.params['status']:
+                self.api_method = method
+
+    def get_monitor_id(self, fname):
+        monitors = self.get_monitors()
+        for m_dict in monitors['monitors']:
+            if m_dict['friendly_name'] == fname:
+                mid = m_dict['id']
+                return mid
+
+    def get_monitors(self):
+        data = {}
+        data['api_key'] = self.params['api_key']
+        data = json.dumps(data)
+        self.uri = API_BASE + "getMonitors"
+        req = Request()
+        state = req.post(url=self.uri, data=data, headers=self.headers)
+        state = state.read()
+        state = json.loads(state)
```

Comment:
```suggestion
        state = json.loads(to_text(state.read(), errors='surrogate_or_strict'))
```

This makes this call future proof.

Reply:
https://github.com/Ninjiner/ansible/commit/febdd0a55d5ccab79aa34687e9cb4c91d4889c08